brainscore_language/plugin_management/test_plugins.py

```diff
@@ -43,17 +43,34 @@ def __call__(self):
         self.teardown()
 
     def validate_plugin(self):
-        """ requires "test.py" file in plugin directory """
         assert (self.plugin_directory / 'test.py').is_file(), "'test.py' not found"
 
+        # if environment.yml is present, ensure no dependency conflicts
+        # checks that environment.yml does not include env name or unsupported python versions
+        conda_yml_path = self.plugin_directory / 'environment.yml'
+        if conda_yml_path.is_file():
+            with open(conda_yml_path, "r") as f:
+                try:
+                    env = yaml.dump(yaml.safe_load(f))
+                    assert 'name' not in env, f"\nenvironment.yml must not specify 'name'"
+                    python_specs = [line for line in env.split("\n") if 'python=' in line]
+                    if len(python_specs) > 0:
+                        if len(python_specs) == 1:
+                            python_spec = python_specs[0]
+                            python_version = python_spec.split('python=')[1]
+                            assert python_version.startswith(('3.7', '3.8', '3.9'))
+                        else:
+                            warnings.warn('multiple versions of python found in environment.yml')
```

Comment:
simplify and raise instead of warning
```suggestion
                    if len(python_specs) == 1:
                        python_spec = python_specs[0]
                        python_version = python_spec.split('python=')[1]
                        assert python_version.startswith(('3.7', '3.8', '3.9'))
                    elif len(python_specs) > 1:
                            raise yaml.YAMLError('multiple versions of python found in environment.yml')
                    # if no python specifications, ignore
```